File: JavaScript/leetcode75/src/test/Problem37_LongestZigZagPathInABinaryTree.test.js
const longestZigZag = require("../main/Problem37_LongestZigZagPathInABinaryTree");

//  Definition for a binary tree node.
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

test("should return longest zig zag path", () => {
  let root = new TreeNode(1, null, 1);
  root.right = new TreeNode(1, 1, 1);
  root.right.left = new TreeNode(1, null, null);
  root.right.right = new TreeNode(1, 1, 1);
  root.right.right.left = new TreeNode(1, null, 1);
  root.right.right.left.right = new TreeNode(1, null, 1);
  root.right.right.right = new TreeNode(1, null, null);

  expect(longestZigZag(root)).toBe(3);
});

File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem37_LongestZigZagPathInABinaryTreeTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;
import org.phoenix.leetcode.challenges.Problem33_MaximumDepthOfBinaryTree.TreeNode;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem37_LongestZigZagPathInABinaryTreeTest {

    private final Problem37_LongestZigZagPathInABinaryTree test = new Problem37_LongestZigZagPathInABinaryTree();
    @Test
    void longestZigZag() {
        TreeNode root = new TreeNode(1);
        root.right = new TreeNode(1);
        root.right.left = new TreeNode(1);
        root.right.right = new TreeNode(1);
        root.right.right.left = new TreeNode(1);
        root.right.right.left.right = new TreeNode(1);
        root.right.right.left.right.right = new TreeNode(1);
        root.right.right.right = new TreeNode(1);


        assertEquals(3, test.longestZigZag(root));
    }
}